providers/configcat/src/main/java/dev/openfeature/contrib/providers/configcat/ConfigCatProviderConfig.java

```diff
@@ -0,0 +1,24 @@
+package dev.openfeature.contrib.providers.configcat;
+
+import com.configcat.ConfigCatClient;
+import lombok.Builder;
+import lombok.Getter;
+
+import java.util.function.Consumer;
+
+
+/**
+ * Options for initializing ConfigCat provider.
+ */
+@Getter
+@Builder
+public class ConfigCatProviderConfig {
+    private Consumer<ConfigCatClient.Options> options;
+
+    // Only holding temporary for initialization
+    private String sdkKey;
+
+    public void postInit() {
+        sdkKey = null; // for security, not holding key in memory for long-term
```

Comment:
I don't really see the benefit of setting this to null. The ConfigCat SDK holds the SDK key anyway and it needs to know about it during the full run.

Reply:
See some [references](https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords/8889285#8889285).

> Some information, such as Social Security numbers (SSNs) and passwords, is highly sensitive. This information should not be kept for longer than necessary nor where it may be seen, even by administrators. For instance, it should not be sent to log files and its presence should not be detectable through searches. Some transient data may be kept in mutable data structures, such as char arrays, and **cleared immediately after use**. Clearing data structures has reduced effectiveness on typical Java runtime systems as objects are moved **in memory** transparently to the programmer.

Reply:
I'm fine with it if you leave it.
But I still think that this is unnecessary as the SDK holds the SDK key anyway (as it needs it for the whole run) so this doesn't provide any layer of extra security but only some extra lines of code to maintain in the long term.

Reply:
In my opinion it is better to leave it, as the SDK implementation can be treated as much as possible as a black-box, without need to know its implementation, or adjust to inner implementation changes. Here the key at the provider config itself is not used, then no need to keep it. I'd leave it for now, thanks.